src/lib/gssapi/krb5/accept_sec_context.c

```diff
@@ -782,6 +783,7 @@ kg_accept_krb5(minor_status, context_handle,
                 goto fail;
             }
 
+            have_cb_success = 1;
```

Comment:
You can just set GSS_C_CHANNEL_BOUND_FLAG right here, and get rid of have_cb_success.  It doesn't need to be conditional on mutual auth or ret_flags_understood.  There should be no reason to unset to flag, since it should just never be set in gss_flags other than here.